File: openHAB/SelectionUITableViewCell.m
//
//  SelectionUITableViewCell.m
//  openHAB
//
//  Created by Victor Belov on 27/01/14.
//  Copyright (c) 2014 Victor Belov. All rights reserved.
//

#import "SelectionUITableViewCell.h"
#import "OpenHABWidget.h"
#import "OpenHABItem.h"
#import "OpenHABWidgetMapping.h"

@implementation SelectionUITableViewCell

- (void)setWidget:(OpenHABWidget *)widget
{
    super.widget = widget;
    self.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    self.selectionStyle = UITableViewCellSelectionStyleBlue;
}

- (void)displayWidget
{
    self.textLabel.text = [self.widget labelText];
    NSUInteger selectedMapping = [self.widget mappingIndexByCommand:self.widget.item.state];
    if (selectedMapping != NSNotFound) {
        OpenHABWidgetMapping *widgetMapping = [widget.mappings objectAtIndex:selectedMapping];
        self.detailTextLabel.text = widgetMapping.label;
    }
}

@end

File: openHAB/SelectionUITableViewCell.swift
// Copyright (c) 2010-2020 Contributors to the openHAB project
//
// See the NOTICE file(s) distributed with this work for additional
// information.
//
// This program and the accompanying materials are made available under the
// terms of the Eclipse Public License 2.0 which is available at
// http://www.eclipse.org/legal/epl-2.0
//
// SPDX-License-Identifier: EPL-2.0

import OpenHABCore

class SelectionUITableViewCell: GenericUITableViewCell {
    override var widget: OpenHABWidget! {
        get {
            super.widget
        }
        set(widget) {
            super.widget = widget
            accessoryType = .disclosureIndicator
            selectionStyle = .blue
        }
    }

    override func displayWidget() {
        super.customTextLabel?.text = widget.labelText
        if let selectedMapping = widget.mappingIndex(byCommand: widget.item?.state) {
            if let widgetMapping = widget?.mappingsOrItemOptions[Int(selectedMapping)] {
                customDetailTextLabel?.text = widgetMapping.label
            }
        } else {
            customDetailTextLabel?.text = ""
        }
    }
}
